powerauth-backend-tests/src/test/java/com/wultra/security/powerauth/test/shared/util/TemporaryKeyFetchUtil.java

```diff
@@ -0,0 +1,200 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.wultra.security.powerauth.test.shared.util;
+
+import com.nimbusds.jose.JWSAlgorithm;
+import com.nimbusds.jose.JWSHeader;
+import com.nimbusds.jose.util.Base64URL;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.SignedJWT;
+import com.wultra.core.rest.client.base.RestClient;
+import com.wultra.security.powerauth.configuration.PowerAuthTestConfiguration;
+import com.wultra.security.powerauth.model.TemporaryKey;
+import io.getlime.core.rest.model.base.request.ObjectRequest;
+import io.getlime.core.rest.model.base.response.ObjectResponse;
+import io.getlime.security.powerauth.crypto.lib.encryptor.model.EncryptorScope;
+import io.getlime.security.powerauth.crypto.lib.generator.KeyGenerator;
+import io.getlime.security.powerauth.crypto.lib.util.HMACHashUtilities;
+import io.getlime.security.powerauth.crypto.lib.util.KeyConvertor;
+import io.getlime.security.powerauth.crypto.lib.util.SignatureUtils;
+import io.getlime.security.powerauth.lib.cmd.consts.PowerAuthVersion;
+import io.getlime.security.powerauth.lib.cmd.util.JsonUtil;
+import io.getlime.security.powerauth.lib.cmd.util.MapUtil;
+import io.getlime.security.powerauth.lib.cmd.util.RestClientFactory;
+import io.getlime.security.powerauth.rest.api.model.request.TemporaryKeyRequest;
+import io.getlime.security.powerauth.rest.api.model.response.TemporaryKeyResponse;
+import org.bouncycastle.asn1.ASN1EncodableVector;
+import org.bouncycastle.asn1.ASN1Integer;
+import org.bouncycastle.asn1.DLSequence;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+
+import javax.crypto.SecretKey;
+import java.math.BigInteger;
+import java.nio.charset.StandardCharsets;
+import java.security.PublicKey;
+import java.security.interfaces.ECPublicKey;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.*;
+
+/**
+ * Utilities for fetching temporary keys from the server.
+ *
+ * @author Roman Strobl, roman.strobl@wultra.com
+ */
+public class TemporaryKeyFetchUtil {
+
+    private TemporaryKeyFetchUtil() {
+    }
+
+    private static final KeyGenerator KEY_GENERATOR = new KeyGenerator();
+    private static final KeyConvertor KEY_CONVERTOR = new KeyConvertor();
+    private static final SignatureUtils SIGNATURE_UTILS = new SignatureUtils();
+
+    /**
+     * Fetch temporary key for encryption from the server.
+     * @param version Protocol version.
+     * @param scope Encryption scope.
+     * @param config Test configuration.
+     * @throws Exception Thrown in case temporary key fetch fails.
+     */
+    public static TemporaryKey fetchTemporaryKey(PowerAuthVersion version, EncryptorScope scope, PowerAuthTestConfiguration config) throws Exception {
+        final RestClient restClient = RestClientFactory.getRestClient();
+        if (restClient == null) {
+            return null;
```

Comment:
🤔 Isn't it better to avoid NPE and return `Optional<TemporaryKey>` instead? Then we may use `.orElse()` or `.map()`.

Reply:
I will remove this code.